lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -78,29 +78,38 @@ def figure_edit(axes, parent=None):
                 continue
             linedict[label] = line
         curves = []
-        linestyles = list(six.iteritems(LINESTYLES))
-        drawstyles = list(six.iteritems(DRAWSTYLES))
-        markers = list(six.iteritems(MARKERS))
+
+        def prepare_data(d, init):
+            """Prepare entry for FormLayout.
+            """
+            # List items in dict, dropping duplicate values, sorting by values.
+            kvs = [(k, v) for v, k in
+                   sorted({v: k for k, v in d.items()}.items())]
```

Comment:
This will be non-deterministic only in the sense that the "nothing" entry (the only one with multiple internal synonyms) may be *internally* represented as one of "nothing", "none", "None" or None depending on which one comes up last in the iteration.  From the UI point of view there is no difference.  See comment above.